### PR TITLE
[CODEGEN-1918] Support concurrent requests for Custom Models

### PR DESCRIPTION
**Pull request by yura-ant**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Allow enable flask process via command line option --max_workers or define CUSTOM_MODEL_WORKERS  numeric runtime parameter
Option 3 from
https://docs.google.com/document/d/1yq-T7-pmjW49RsvbslNIl3BF_GRbYLti2McVmZgd_To/edit#heading=h.8vln2nt42rgh

TODO
I would like to add more test. But in general I think it is ready for review

### CODE REVIEW

**Comment by yura-ant** [comment]
jarvis please test failed

**Comment by yura-ant** [comment]
jenkins run test_functional_by_framework-check_by_framework

File: tests/functional/test_inference_per_framework.py [hunk 1]
```diff
@@ -399,6 +406,7 @@ def test_custom_models_with_drum_prediction_server(
         ],
     )
     @pytest.mark.parametrize("pass_args_as_env_vars", [True])
+    @pytest.mark.parametrize("max_workers", [1])
```
Comment (baekdahl) [inline tests/functional/test_inference_per_framework.py:409]: Why parametrize when there is only one param?

**Review by baekdahl — APPROVED** [review]
LGTM.

File: tests/functional/test_inference_per_framework.py (hunk 1 above)
Comment (yura-ant) [reply tests/functional/test_inference_per_framework.py:409]: To keep it consistent with @pytest.mark.parametrize("pass_args_as_env_vars", [True]). As we reuse same function in different tests. I think the test is too much parameterise but it works

**Review by scottp-ml — APPROVED** [review]
LGTM

**Review by eric-s-s — APPROVED** [review]

**Comment by yura-ant** [comment]
Hi @yakov-g,  Have you had a chance to look at this?

**Review by klichukb — APPROVED** [review]
Good start and no nginx/uwsgi stack!

Overall for the future I think we might still consider using gunicorn for example, as I think app.run(processes) is still [not suggested ](https://flask.palletsprojects.com/en/3.0.x/deploying/) as a production-grade option for serving multiple clients

**Pull request merged**